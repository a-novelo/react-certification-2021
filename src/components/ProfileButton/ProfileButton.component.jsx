import React, { useState } from 'react';
import { useHistory } from 'react-router';
import { faUserCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  ProfileButtonWrapper,
  ProfileButtonContainer,
  UserAvatar,
} from './ProfileButton.styled';
import { useAuth } from '../../providers/Auth';
import LoginModal from '../LoginModal/LoginModal.component';

function ProfileButton() {
  const [showLoginModal, setShowLoginModal] = useState(false);
  const { getUser, logout } = useAuth();
  const history = useHistory();

  const userData = getUser();
  const userAvatar = userData ? userData.avatarUrl : '';

  function deAuthenticate(event) {
    event.preventDefault();
    logout();
    history.push('/');
  }

  return (
    <ProfileButtonWrapper>
      <LoginModal
        showModal={showLoginModal}
        handleClose={() => {
          setShowLoginModal(false);
        }}
      />
      <ProfileButtonContainer>
        {userAvatar === '' ? (
          <FontAwesomeIcon
            icon={faUserCircle}
            data-testid="profile-icon"
            size="2x"
            onClick={() => {
              setShowLoginModal(true);
            }}
          />
        ) : (
          <UserAvatar id="imagenAjustador" src={userAvatar} onClick={deAuthenticate} />
        )}
      </ProfileButtonContainer>
    </ProfileButtonWrapper>
  );
}

export default ProfileButton;
