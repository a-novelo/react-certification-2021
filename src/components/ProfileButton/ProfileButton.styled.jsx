import styled from 'styled-components';

const Wrapper = styled.button`
  background: none;
  border: none;
  color: inherit;
  cursor: pointer;
  display: flex;
  font: inherit;
  margin-left: auto;
  outline: inherit;
  padding: 0;
`;

export { Wrapper };
