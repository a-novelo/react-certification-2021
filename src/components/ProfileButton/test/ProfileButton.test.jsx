import React from 'react';
import renderer from 'react-test-renderer';
import ProfileButton from '../ProfileButton.component';

describe('#ProfileButton', () => {
  it('displays correctly', () => {
    const tree = renderer.create(<ProfileButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
