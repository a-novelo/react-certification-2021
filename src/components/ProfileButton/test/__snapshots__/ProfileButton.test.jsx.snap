// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProfileButton displays correctly 1`] = `
<div
  className="sc-bdfBwQ"
>
  <div
    className="sc-hKgILt dcQLHf"
  >
    <div
      className="sc-eCssSg kMjJFf"
    >
      Login
      <div
        className="sc-jSgupP fNfaBT"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-exclamation-circle fa-w-16 "
          data-icon="exclamation-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Invalid username and/or password
      </div>
      <form
        className="sc-gKsewC ytCje"
        onSubmit={[Function]}
      >
        <div
          className="sc-iBPRYJ bYjIqV"
        >
          <label
            className="sc-fubCfw"
            htmlFor="username"
          >
            <strong
              className="sc-pFZIQ eQswuU"
            >
              username 
            </strong>
            <input
              className="sc-jrAGrp bNwOnU"
              id="username"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          className="sc-iBPRYJ bYjIqV"
        >
          <label
            className="sc-fubCfw"
            htmlFor="password"
          >
            <strong
              className="sc-pFZIQ eQswuU"
            >
              password 
            </strong>
            <input
              className="sc-jrAGrp bNwOnU"
              id="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
          </label>
        </div>
        <div
          className="sc-kEjbxe"
        >
          <button
            className="sc-iqHYGH eKSCrJ"
            onClick={[Function]}
            type="button"
          >
            CANCEL
          </button>
          <button
            className="sc-iqHYGH eKSCrJ"
            type="submit"
          >
            LOGIN
          </button>
        </div>
      </form>
    </div>
  </div>
  <button
    className="sc-gsTCUz EPOPm"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-user-circle fa-w-16 fa-2x "
      data-icon="user-circle"
      data-prefix="fas"
      data-testid="profile-icon"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
</div>
`;
